mlflow/pyfunc/__init__.py

```diff
@@ -486,6 +486,67 @@ def load_model(
     return PyFuncModel(model_meta=model_meta, model_impl=model_impl, predict_fn=predict_fn)
 
 
+class ModelServerPyfunc:
```

Comment:
Nit name adjustment suggestion (in any case, we should prefix the model with `_` so that we don't include it in the public docs). Can we also inherit from `PyFuncModel` and add model metadata to the class, which we can retrieve from the model directory's `MLmodel` file?

```suggestion
class _ServedPyFuncModel(PyFuncModel)
```